extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/config/TransitKeyConfig.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.vault.runtime.config;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class TransitKeyConfig {
+
+    /**
+     * Specifies the name of the key to use. By default this will be the property key alias. Used when
+     * the same transit key is used with different configurations. Such as:
+     * 
+     * <pre>
+     *     {@code
+     *     quarkus.vault.transit.key.foo1.name=foo
+     *     quarkus.vault.transit.key.foo1.key_version=1
+     *     quarkus.vault.transit.key.foo2.name=foo
+     *     quarkus.vault.transit.key.foo2.key_version=2
+     *     }
+     * </pre>
+     */
+    @ConfigItem
+    public Optional<String> name;
+
+    // sign
+
+    /**
+     * Set to true when the input is already hashed.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#prehashed">api documentation for prehashed</a>
+     */
+    @ConfigItem
+    public Optional<Boolean> prehashed;
+
+    /**
+     * When using a RSA key, specifies the RSA signature algorithm.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#signature_algorithm">api documentation for
+     *      signature_algorithm</a>
+     */
+    @ConfigItem
```

Comment:
Perhaps we can be opinionated and set the default to `pkcs1v15`. It is the value which is also recommended [here](https://tools.ietf.org/html/rfc7518#section-3.1) for JWS. We can leave it optional, just a suggestion

Reply:
no strong opinion here. I do not have enough knowledge in that field (such as common usages). as a result I tend to stay close to the api (ie: in that case keep the default, which means not setting any value, which will result in `pss` according to doc).
let me know if you feel we should really change that.

Reply:
@vsevel, that is OK, lets  keep the default value